drivers/i2s/i2s_ll_stm32.c

```diff
@@ -30,6 +30,16 @@ static unsigned int div_round_closest(uint32_t dividend, uint32_t divisor)
 	return (dividend + (divisor / 2U)) / divisor;
 }
 
+static bool queue_is_empty(struct ring_buf *rb)
+{
```

Comment:
This function is vulnerable to race conditions - while this isn't an issue ***now*** because all callers already suspend IRQs, it could become one in the future.
Please either:
* add a comment indicating this function must be called with IRQ suspended, or
* suspend interrupts for the duration of the function with an `irq_lock`/`irq_unlock` pair

Nit: this really should be a lock on the `struct ring_buf` instead of merely IRQ suspensions; this is acceptable(ish) because only one CPU at a time may interact with the DMA/I2S peripheral.